src/main/resources/application.conf

```diff
@@ -53,29 +53,51 @@ mantis {
       # Turn discovery of/off
       discovery-enabled = true
 
+      # Externally visible hostname or IP.
+      host = ""
+
       # Listening interface for discovery protocol
       interface = "0.0.0.0"
 
       # Listening port for discovery protocol
       port = 30303
 
-      # Maximum discovered nodes stored (TODO: remove me once full protocol is in place)
-      nodes-limit = 5000
-
-      # Initial delay for discovery scan
-      scan-initial-delay = 20.seconds
+      # If true, the node considers the bootstrap and the previously persisted nodes
+      # as already discovered and uses them as peer candidates to get blocks from.
+      # Otherwise it enroll with the bootstrap nodes and gradually discover the
+      # network every time we start, eventually serving candidates.
+      #
+      # Useful if discovery has problem, as the node can start syncing with the
+      # bootstraps straight away.
+      #
+      # Note that setting reuse-known-nodes and discovery-enabled to false at the
+      # same time would mean the node would have no peer candidates at all.
+      reuse-known-nodes = true
```

Comment:
Sorry I couldn't find a good name for it. It's basically whether it should work as Mantis does now (`true`), that is, whether it should always treat nodes the `bootstrapNodes` and potentially the ones in `knownNodesStorage` as if they passed the discovery process and give it back to the `PeerManagerActor` when it asks, even if they may be unreachable. It's fine since that actor handles this case. Or whether we should always go through the discovery process with them, verifying their ENR, and only after that serve them to the `PeerManagerActor`.

Reply:
> And why giving the user the two alternatives? In which case one would be used over the other?

For example the alternative where you don't connect to old nodes was good to be able to actually test how quickly discovery works without distorting it by reconnecting to past nodes. It may also be used in cases where you want to use discovery to re-validate the ENR records, rather than rely on nodes persisted before, for example if you have a list of bootstrap nodes which you don't implicitly trust is correct.

> My question if not is: why making this configurable? If it's set here to true we most likely will keep it that way forever

I agree it will most likely never be set to anything, especially since it seems to take absolute ages to discover nodes that are willing to take on more connections. Nevertheless it seemed rather arbitrary that discovery has to validate the ENR signatures and all, while we implicitly trust and old list of nodes.

Reply:
We can keep the option then... though I see it probably going away after some time